biorbd_optim/problem_type.py

```diff
@@ -97,7 +107,10 @@ def muscle_activations_and_torque_driven(ocp, nlp):
         nlp["nu"] = nlp["u"].rows()
         nlp["var_controls"]["muscles"] = nlp["nbMuscle"]
 
-        ProblemType.__configure_forward_dyn_func(ocp, nlp, Dynamics.forward_dynamics_torque_muscle_driven)
+        if "dynamic" in nlp["problem_type"]:
+            Problem.configure_forward_dyn_func(ocp, nlp, nlp["problem_type"]["dynamic"])
+        else:
+            Problem.configure_forward_dyn_func(ocp, nlp, Dynamics.forward_dynamics_torque_muscle_driven)
```

Comment:
Is there a reason to only have this if for this particular condition and not everywhere?